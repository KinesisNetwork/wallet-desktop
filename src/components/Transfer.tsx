import * as React from 'react'

import { PayeeForm } from '@containers/PayeeForm'
import { TransferForm } from '@containers/TransferForm'
import { TransferView } from '@types'

export interface Props {
  transferView: TransferView
}
export const Transfer: React.SFC<Props> = ({ transferView }) => {
  return (
    <React.Fragment>
      <h1 className="sub-heading primary-font">
        <span style={{ paddingRight: '5px' }}>Transfer</span>
        <span className="has-text-primary">KAU</span>
      </h1>
      {transferView === TransferView.transfer && <TransferForm />}
      {transferView === TransferView.addPayee && <PayeeForm />}
      <div
        className="is-divider is-hidden-tablet"
        style={{ marginBottom: 0, borderTopWidth: '0.01rem' }}
      />
    </React.Fragment>
  )
}

Transfer.displayName = 'Transfer'
