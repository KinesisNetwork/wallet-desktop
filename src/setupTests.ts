import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
