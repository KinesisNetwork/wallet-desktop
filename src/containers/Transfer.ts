import { connect } from 'react-redux'

import { changeTransferView } from '@actions'
import { Transfer as TransferPresentation } from '@components/Transfer'
import { RootState } from '@store'

const mapStateToProps = ({ view: { transferView } }: RootState) => ({
  transferView,
})

const mapDispatchToProps = {
  changeTransferView,
}

export const Transfer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(TransferPresentation)
