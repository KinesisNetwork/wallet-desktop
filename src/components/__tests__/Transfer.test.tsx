import { shallow } from 'enzyme'
import * as React from 'react'

import { Transfer } from '@components/Transfer'
import { TransferView } from '@types'
import '../../setupTests'

describe('Transfer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Transfer transferView={TransferView.transfer} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders a primary heading', () => {
    const wrapper = shallow(<Transfer transferView={TransferView.transfer} />)

    const heading = wrapper.find('h1')

    expect(
      heading
        .find('span')
        .at(0)
        .text(),
    ).toEqual('Transfer')
    expect(
      heading
        .find('span')
        .at(1)
        .text(),
    ).toEqual('KAU')
  })

  it('displays a TransferForm if transfer view indicates', () => {
    const wrapper = shallow(<Transfer transferView={TransferView.transfer} />)

    expect(wrapper.find('Connect(TransferForm)')).toHaveLength(1)
  })
})
