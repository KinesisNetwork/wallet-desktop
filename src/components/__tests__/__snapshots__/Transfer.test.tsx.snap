// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer renders correctly 1`] = `
<Fragment>
  <h1
    className="sub-heading primary-font"
  >
    <span
      style={
        Object {
          "paddingRight": "5px",
        }
      }
    >
      Transfer
    </span>
    <span
      className="has-text-primary"
    >
      KAU
    </span>
  </h1>
  <Connect(TransferForm) />
  <div
    className="is-divider is-hidden-tablet"
    style={
      Object {
        "borderTopWidth": "0.01rem",
        "marginBottom": 0,
      }
    }
  />
</Fragment>
`;
